app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Good, but could be greater. 

This block gets rendered only if the user is authorized per the user#show policy, but the whole page is also rendered per the user#show policy. So effectively, this conditional here is never `false`. How can we fix it?